cxplat/src/cxplat_winuser/workitem_winuser.cpp

```diff
@@ -77,6 +69,9 @@ cxplat_allocate_preemptible_work_item(
 void
 cxplat_queue_preemptible_work_item(_Inout_ cxplat_preemptible_work_item_t* io_work_item)
 {
+    if (!cxplat_acquire_rundown_protection(&_cxplat_preemptible_work_items_rundown_reference)) {
```

Comment:
Does this match Windows kernel behavior?

Reply:
It's equivalent, and required to allow the free to happen after waiting on the work item(s) to complete

Reply:
It's also slightly more performant now in the case where you alloc/free but never queue the work item, since you don't acquire/release rundown protection in that case.